node/podcontroller.go

```diff
@@ -304,6 +305,26 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) (retErr er
 			if key, err := cache.MetaNamespaceKeyFunc(newPod); err != nil {
 				log.G(ctx).Error(err)
 			} else {
+				obj, ok := pc.knownPods.Load(key)
+				if !ok {
+					// Pods are only ever *added* to knownPods in the above AddFunc, and removed
+					// in the below *DeleteFunc*
+					panic("Pod not found in known pods. This should never happen.")
+				}
+
+				kPod := obj.(*knownPod)
+				kPod.Lock()
+				if kPod.lastPodStatusUpdateSkipped && !cmp.Equal(newPod.Status, kPod.lastPodStatusReceivedFromProvider.Status) {
+					// The last pod from the provider -> kube api server was skipped, but we see they no longer match.
```

Comment:
A scene is as follow: 

1. Pod in K8s status A
Pod in Provider status A

2. Pod  in Provider Status change to B
Pod in K8s synced to B

3. Pod in K8s changed to C due to some resom.

Here `lastPodStatusUpdateSkipped` is false. So  we would equeue pod in Provider.

IMO, we can remove  `lastPodStatusUpdateSkipped `

Reply:
I don't think we should do that, otherwise it will lead to a retry storm. If something outside of VK updates the pod status, it will cause use to update the api server, and if that thing tries to restore the status change, we will do it again.

Reply:
Thanks for your explanation!